Tests/SwiftFusionTests/Inference/ChordalInitializationTests.swift

```diff
@@ -0,0 +1,144 @@
+let x3 = TypedID<Pose3, Int>(3)
+
+// ground truth
+let  p0 = Vector3(0,0,0);
+let  R0 = Rot3.fromTangent(Vector3(0.0,0.0,0.0))
+let  p1 = Vector3(1,2,0)
+let  R1 = Rot3.fromTangent(Vector3(0.0,0.0,1.570796))
+let  p2 = Vector3(0,2,0)
+let  R2 = Rot3.fromTangent(Vector3(0.0,0.0,3.141593))
+let  p3 = Vector3(-1,1,0)
+let  R3 = Rot3.fromTangent(Vector3(0.0,0.0,4.712389))
+
+let pose0 = Pose3(R0,p0)
+let pose1 = Pose3(R1,p1)
+let pose2 = Pose3(R2,p2)
+let pose3 = Pose3(R3,p3)
+
+/// simple test graph for the chordal initialization
+func graph1() -> FactorGraph {
+  var g = FactorGraph()
+  g.store(BetweenFactor3(x0, x1, between(pose0, pose1)))
+  g.store(BetweenFactor3(x1, x2, between(pose1, pose2)))
+  g.store(BetweenFactor3(x2, x3, between(pose2, pose3)))
+  g.store(BetweenFactor3(x2, x0, between(pose2, pose0)))
+  g.store(BetweenFactor3(x0, x3, between(pose0, pose3)))
+  g.store(PriorFactor3(x0, pose0))
+  return g
+}
+
+class ChordalInitializationTests: XCTestCase {
+  /// make sure the derivatives are correct
+  func testFrobeniusRot3BetweenJacobians() {
+    
+    print(Matrix3.standardBasis)
+    
+    var val = VariableAssignments()
+    let p0 = val.store(Matrix3(1,0,0,0,1,0,0,0,1))
+    let p1 = val.store(Matrix3(1,0,0,0,1,0,0,0,1))
+    
+    let frf = RelaxedRotationFactorRot3(p0, p1, Matrix3(0.0,0.1,0.2,1.0,1.1,1.2,2.0,2.1,2.2))
+    
+    let frf_j = frf.linearized(at: Tuple2(val[p0], val[p1]))
+    
+    let Rij = Matrix3(0.0,0.1,0.2,1.0,1.1,1.2,2.0,2.1,2.2)
+    let M9: Jacobian9x3x3_2 = Array9([
+      Tuple2(Matrix3(-1,0,0,0,0,0,0,0,0), Matrix3(Rij[0, 0],Rij[0, 1],Rij[0, 2],0,0,0,0,0,0)),
+      Tuple2(Matrix3(0,-1,0,0,0,0,0,0,0), Matrix3(Rij[1, 0],Rij[1, 1],Rij[1, 2],0,0,0,0,0,0)),
+      Tuple2(Matrix3(0,0,-1,0,0,0,0,0,0), Matrix3(Rij[2, 0],Rij[2, 1],Rij[2, 2],0,0,0,0,0,0)),
+      Tuple2(Matrix3(0,0,0,-1,0,0,0,0,0), Matrix3(0,0,0,Rij[0, 0],Rij[0, 1],Rij[0, 2],0,0,0)),
+      Tuple2(Matrix3(0,0,0,0,-1,0,0,0,0), Matrix3(0,0,0,Rij[1, 0],Rij[1, 1],Rij[1, 2],0,0,0)),
+      Tuple2(Matrix3(0,0,0,0,0,-1,0,0,0), Matrix3(0,0,0,Rij[2, 0],Rij[2, 1],Rij[2, 2],0,0,0)),
+      Tuple2(Matrix3(0,0,0,0,0,0,-1,0,0), Matrix3(0,0,0,0,0,0,Rij[0, 0],Rij[0, 1],Rij[0, 2])),
+      Tuple2(Matrix3(0,0,0,0,0,0,0,-1,0), Matrix3(0,0,0,0,0,0,Rij[1, 0],Rij[1, 1],Rij[1, 2])),
+      Tuple2(Matrix3(0,0,0,0,0,0,0,0,-1), Matrix3(0,0,0,0,0,0,Rij[2, 0],Rij[2, 1],Rij[2, 2]))
+    ])
+    
+    let b = Matrix3(0, 0, 0, 0, 0, 0, 0, 0, 0)
+    
+    let jf = JacobianFactor9x3x3_2(jacobian: M9, error: b, edges: Tuple2(TypedID<Matrix3, Int>(0), TypedID<Matrix3, Int>(1)))
+    
+    assertEqual(
+      Tensor<Double>(stacking: frf_j.jacobian.map { $0.tensor }),
+      Tensor<Double>(stacking: jf.jacobian.map { $0.tensor })
+      , accuracy: 1e-4
+    )
+    
+    let fpf = RelaxedAnchorFactorRot3(p0, Matrix3(0.0,0.1,0.2,1.0,1.1,1.2,2.0,2.1,2.2))
+    
+    let fpf_j = fpf.linearized(at: Tuple1(val[p0]))
+    
+    let I_9x9: Jacobian9x3x3_1 = Array9([
```

Comment:
I would feel more convinced with numerical differentiation tests a la GTSAM, that internally use the move(along) operator. The above is also pretty ugly and we don’t know where it comes from.